lib/puppet/indirector/s3.rb

```diff
@@ -0,0 +1,47 @@
+require 'puppet/file_serving/terminus_helper'
+require 'puppet/indirector/terminus'
+require 'aws-sdk-core' if Puppet.features.aws?
```

Comment:
I'm pretty sure that `confine` is specific to providers.
I rather like the approach of querying for a feature.